### PR TITLE
[skip ci]Topology2: Add sof-mtl-cs42l43-l0-cs35l56-l23

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Add sof-mtl-cs42l43-l0-cs35l56-l23 topology to
support a variant configuration with Cirrus Logic codecs.

Not tested yet

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
@charleskeepax fyi

**Review by charleskeepax — APPROVED** [review]
We don't have any MTLs to test this, but it looks fine to me.

**Comment by aiChaoSONG** [comment]
@charleskeepax Could you please point out where is the UCM for cs42l43 and cs35l56? I'd like it to be verified before submitting the official kernel and topology patches.

**Comment by charleskeepax** [comment]
The cs35l56 bits are merged here:
https://github.com/alsa-project/alsa-ucm-conf/commit/8e39917518b62cce44d098bd6fe5d2c1b41f92bd
The cs42l43 are still pending review from the maintainer:
https://lore.kernel.org/alsa-devel/20231220165951.3675985-1-ckeepax@opensource.cirrus.com/

If you guys could provide a reviewed-by or tested-by, I am sure that would help move things along. It is probably time I did a resend on the cs42l43 stuff, so I will do that today and if you guys could put any review comments on that would be greatly appreciated.

**Review by lgirdwood — APPROVED** [review]
@aiChaoSONG looks like we are good once you have validated with UCM.

**Comment by charleskeepax** [comment]
Ok, Jaroslav just applied the patch as soon as I resent it (apologies thought I posted this comment days ago but it turns out it was just sitting here):
https://github.com/alsa-project/alsa-ucm-conf/commit/e87dde51d68950537f92af955ad0633437cc419a
So that should be all the UCM required for the pending cs35l56/cs42l43 systems. Worth keeping the caveat about sequential amps in my email on the 15th in mind however.

**Comment by lgirdwood** [comment]
@aiChaoSONG some conflicts, are all dependencies merged ?

**Comment by charleskeepax** [comment]
I believe this can be abandoned, this has been merged through another patch: eea7b51fd67d ("topology2: cavs-cs42l43: Add topology for cs42l43/cs35l56 on MTL")

**Pull request closed without merging**
